### DISCUSSION TITLE
How to handle client side cancel the request?

### DISCUSSION THREAD

**Discussion by tom8zds** [tokio-rs/axum#2811] [Q&A]
### Summary

Need help. 

I tried all popular rust http framework, but it seems no one can handle the clientside closing the connection.

what I want to do is restore the data change when user close the request.

Is there any way I can achieve this.

```rust
async fn handle() {
   add_sth();
   select! {
      _ = do_sth() => {
          return OK();
      },
      _ = check_cancel() => {
          restore_add();
      }
   }
}
```

### axum version

0.7.3

**Comment by mladedav** [comment]
The future is cancelled when the connection closes. If you want it to survive, you need to use something like `tokio::spawn` inside the handler.

You can use channels to pass around information and a drop guard in the handler to signal that the future has been dropped and the request cancelled.

**Reply by tom8zds** [reply]
Thank you for help, I managed to accomplish the work.
I have tried the drop guard the day before yesterday but by some unkonwn reason it does not work. 
Just after opening the discussion I tried a new minimal demo it works fine.  😂
